js/AbstractPropertiesList.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright 2017-present, The Visdom Authors
+ * All rights reserved.
+ *
+ * This source code is licensed under the license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import React from 'react';
+
+class MyRef {
```

Comment:
Seeing this again suggests that someday it'll be worthwhile to update to hook-based react. Certainly has made a lot of my more recent projects tractable. Unsure if it's something we can really adopt gradually though.

Reply:
I've just read the reference of the react hooks API. I would like to rewrite some of the easier classes to the new API, but I suggest to create new PR for this for simplicity reasons.

Reply:
Agreed - not important for this PR, but may make working in this codebase easier for you and everyone moving forward!